packages/core/src/controllers/store.ts

```diff
@@ -40,6 +52,11 @@ export class Store<Key, Data extends StoreStruct> extends IStore<Key, Data> {
         } else if (isSessionStruct(value)) {
           // TODO(pedro) revert type casting as any
           this.map.set(value.topic as any, value);
+        } else if (this.getKey) {
+          // Null checking to prevent undefined or null values to throw an error
```

Comment:
Could all of this be combined into `else if (this.getKey && value)` ? ideally also used alongside `isUndefined` util to prevent truthy values slipping by